lib/llvm-backend/src/state.rs

```diff
@@ -240,12 +240,15 @@ impl State {
         &self,
         n: usize,
     ) -> Result<&[(BasicValueEnum, ExtraInfo)], BinaryReaderError> {
-        self.stack
-            .get(self.stack.len() - n..)
-            .ok_or(BinaryReaderError {
+        if self.stack.len() < n {
+            return Err(BinaryReaderError {
                 message: "invalid value stack",
                 offset: -1isize as usize,
-            })
+            });
+        }
+
+        let new_len = self.stack.len() - n;
+        Ok(&self.stack[new_len..])
```

Comment:
Minor style feedback:

⛳️ 
```Rust
let new_len = self.stack.len().checked_sub(n).ok_or( /* error code goes here */ );
// do other stuff here
```

I think this is a bit more explicit and concise!

This can be combined into a singe statement, but either style is fine.

Reply:
I agree with you, it would has been more clear in your way but the code will more complicated at the end because either:
* `wasmparser::primitives:BinaryReaderError` doesn't implement `convert::From`
* `warning: unreachable expression`
* unused varaible will be detected by the compiler somehow so we will need to rename `new_len` to `_new_len`.

Reply:
Hmm, sorry I'm not following, I'll write out fully what I meant:

I think the entire function can just be:

```Rust
pub fn peekn_extra(&self, n: usize) -> Result<&[(BasicValueEnum, ExtraInfo)], BinaryReaderError> {
   let new_len = self.stack.len().checked_sub(n)
            .ok_or(BinaryReaderError {
                message: "invalid value stack",
                offset: -1isize as usize,
            });
   Ok(&self.stack[new_len..])
}
```

Let me know if that doesn't make sense

Reply:
I've written the same and got some error but my bad i forgot to propagate the Err ;)